src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectory.Stream.Tests.cs

```diff
@@ -359,8 +361,13 @@ private void ExtractToDirectory_ExactRootDirMatch_RegularFile_And_Directory_Thro
             string entryFolderName = inverted ? "folderSibling" : "folder";
             string destinationFolderName = inverted ? "folder" : "folderSibling";
 
+#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+            // the folder used to store files needs to have a shorter path on Apple mobile platforms,
+            // because the TempDirectory gets created in folder with a path longer than 100 bytes
+            using TempDirectory root = new TempDirectory("/Users/helix-runner/Library/Developer/CoreSimulator/Devices/tempDir");
```

Comment:
The `/Users/helix-runner/Library/Developer/CoreSimulator/Devices/tempDir` string is 68 chars long. In the tests where we use them, feel free to reduce the names of the files to even shorter lengths if needed (let's see if the CI does not complain). For example, instead of `file.txt`, you can just use `f`.

Reply:
I will try to shrink the path even more. One concern regarding this is permissions and possibility to create a particular directory.